### PR TITLE
Add `labelalias` to various axes

### PR DESCRIPTION
**Pull request by archmoj**
@plotly/plotly_js

### CODE REVIEW

File: src/plots/cartesian/layout_attributes.js [hunk 1]
```diff
@@ -633,6 +633,17 @@ module.exports = {
...
     },
+    labelalias: {
+        valType: 'any',
+        dflt: false,
+        editType: 'ticks',
+        description: [
+            'Determines the alias for certain labels for ticks and hover.',
+            'For example using {US: \'USA\', CA: \'Canada\'} maps US to USA',
+            'and CA to Canada if tick or hover labels match US or Canada i.e.',
+            'after adding prefix and suffix to labels.'
```
Comment (alexcjohnson) [inline src/plots/cartesian/layout_attributes.js:644]:
```suggestion
            'Replacement text for specific tick or hover labels.',
            'For example using {US: \'USA\', CA: \'Canada\'} changes US to USA',
            'and CA to Canada. The labels we would have shown must match',
            'the keys exactly, after adding any tickprefix or ticksuffix.',
            'labelalias can be used with any axis type, and both keys (if needed)',
            'and values (if desired) can include html-like tags or MathJax.'
```

And to that last point, can we include some tags (`ZERO<sup>2</sup>` or something) and math in a couple of the updated mocks?

**Review by alexcjohnson — APPROVED** [review]
💃 Great to see how compact the code change is! After considering my comment https://github.com/plotly/plotly.js/pull/6481/files#r1115892864 this looks ready to me 🚀

**Comment by archmoj** [comment]
There might be some limitations related to inherited object keys e.g. `__proto__`.
@alexcjohnson do you we should test for something like this?

```json 
{
    "constructor": "my constructor",
    "hasOwnProperty": "my hasOwnProperty",
    "isPrototypeOf": "my isPrototypeOf",
    "propertyIsEnumerable": "my propertyIsEnumerable",
    "toLocaleString": "my toLocaleString",
    "toString": "my toString",
    "valueOf": "my valueOf",
    "__defineGetter__": "my __defineGetter__",
    "__defineSetter__": "my __defineSetter__",
    "__lookupGetter__": "my __lookupGetter__",
    "__lookupSetter__": "my __lookupSetter__",
    "__proto__": "my __proto__",
    "get __proto__": "my get __proto__",
    "set __proto__": "my set __proto__"
}
``` 

Also there may be similar limitations in python dictionaries? cc: @nicolaskruchten

File: test/image/mocks/polar_blank.json [hunk 2]
```diff
@@ -33,6 +33,9 @@
         "x": [0, 0.46],
         "y": [0.56, 1]
       },
+      "angularaxis": {
+        "labelalias": { "0": "ZERO<sup>2</sup>" }
```
Comment (archmoj) [inline test/image/mocks/polar_blank.json:37]: Still I am not 100% sure about this one in polar.
Shouldn't we use the `0°` key instead of `0`?

File: test/image/mocks/ternary-mathjax.json [hunk 3]
```diff
@@ -39,7 +39,15 @@
...
         "showline": true,
-        "showgrid": true
+        "showgrid": true,
+        "labelalias": {
+          "0": "$0^2$",
+          "20": "$20^2$",
+          "40": "$40^2$",
+          "60": "$60^2$",
+          "80": "$80^2$",
+          "100": "$100^2$"
```
Comment (archmoj) [inline test/image/mocks/ternary-mathjax.json:49]: Wondering if one could possibly add `$` in prefix and suffix to render every tick by mathjax?

File: src/plots/cartesian/layout_attributes.js (hunk 1 above)
Comment (archmoj) [reply src/plots/cartesian/layout_attributes.js]: Addressed in [2b75c3c](https://github.com/plotly/plotly.js/pull/6481/commits/2b75c3cf1ad4ca5eb8ac35951971f62b47b2980f) and [c83ac53](https://github.com/plotly/plotly.js/pull/6481/commits/c83ac53801e14db81ef1e0fef0d9269da116015c).

**Comment by archmoj** [comment]
> 428096e

Addressed by [428096e](https://github.com/plotly/plotly.js/pull/6481/commits/428096e8ad3310cfc45903808db9477c32dce3ac).

**Comment by archmoj** [comment]
> Oh that's a good point - I don't think any of those built-in keys will be a string unless you do something truly strange... but it wouldn't hurt to tighten up the initial condition `if(ax.labelalias)` into perhaps:
> 
> ```js
> if(Lib.isPlainObject(ax.labelalias) && ax.labelalias.hasOwnProperty(out.text))
> ```
> 
> Or maybe even better, move the `isPlainObject` test into supplyDefaults (ie delete `containerOut.labelalias` if it's not a plain object) so we don't have to repeat that all the time.

Addressed by [428096e](https://github.com/plotly/plotly.js/pull/6481/commits/428096e8ad3310cfc45903808db9477c32dce3ac).

**Review by alexcjohnson — APPROVED** [review]
🌟 💃

File: test/image/mocks/polar_blank.json (hunk 2 above)
Comment (archmoj) [reply test/image/mocks/polar_blank.json:37]: @alexcjohnson Is this case correct?

File: test/image/mocks/ternary-mathjax.json (hunk 3 above)
Comment (alexcjohnson) [reply test/image/mocks/ternary-mathjax.json:49]: Not sure why you would do that, but maybe it would work? Not particularly related to this PR though...

File: test/image/mocks/polar_blank.json (hunk 2 above)
Comment (archmoj) [reply test/image/mocks/polar_blank.json]: Fixed by [988cde1](https://github.com/plotly/plotly.js/pull/6481/commits/988cde1a8e17d0098fa26b011a9cedee6e99109e).

File: test/image/mocks/ternary-mathjax.json (hunk 3 above)
Comment (archmoj) [reply test/image/mocks/ternary-mathjax.json]: To have dynamic mathjax labels when you zoom in & out. 
Illustrated in [c1d127e](https://github.com/plotly/plotly.js/pull/6481/commits/c1d127e98f6b0411fd7c5c884861ef4ea87c8266).
